### PR TITLE
Use mouse to select files in file opener

### PR DESCRIPTION
**Pull request by elliette**
Fixes https://github.com/flutter/devtools/issues/3757

Fixes an issue where using the mouse to select a file in the file-opener was not working. This was because the tile contents used to be a simple `TextSpan`, and so getting the value with the `text` getter returned the correct value. However, when we switched to having the tile contents be made up of highlighted and unhighlighted `TextSpans`, the `text` getter no longer returned the right value because the getter does not return the values of children (see [documentation](https://api.flutter.dev/flutter/painting/TextSpan/text.html)). This was resolved by using the `getPlainText()` method instead. 

I've also added a `MouseRegion` to detect when a tile is hovered, and switched to having the hovered index as a `ValueListenable` so that we can rebuild when the hovered index is changed. Finally, I created a `StateLessWidget` for the autocomplete tiles, to make it easier to add a test that tapping one of them selects the correct value. 

**Fixed behavior**

mouse_file_search

**Previous behavior**

file_search_before2

### CODE REVIEW

File: packages/devtools_app/lib/src/ui/search.dart [hunk 1]
```diff
@@ -395,6 +381,65 @@ class AutoCompleteState extends State<AutoComplete> with AutoDisposeMixin {
...
+          controller.selectTheSearch = true;
+          controller.search = selected;
+          onTap(selected);
+        },
+        child: ValueListenableBuilder(
+          valueListenable: controller.currentDefaultIndexListenable,
+          builder: (
+            context,
+            currentDefaultIndex,
+            _,
```
[the remark this answers is not in the document]
Comment (elliette) [reply packages/devtools_app/lib/src/ui/search.dart]: Done, thanks!

File: packages/devtools_app/lib/src/ui/search.dart [hunk 2]
```diff
@@ -395,6 +381,65 @@ class AutoCompleteState extends State<AutoComplete> with AutoDisposeMixin {
...
+          ) {
+            return Container(
```
[the remark this answers is not in the document]
Comment (elliette) [reply packages/devtools_app/lib/src/ui/search.dart:418]: Oh nice, done!

File: packages/devtools_app/lib/src/ui/search.dart [hunk 3]
```diff
@@ -456,7 +501,16 @@ mixin AutoCompleteSearchControllerMixin on SearchControllerMixin {
 
   OverlayEntry? autoCompleteOverlay;
 
-  int currentDefaultIndex = 0;
+  ValueListenable<int> get currentDefaultIndexListenable =>
+      _currentDefaultIndex;
+
+  int get currentDefaultIndex => _currentDefaultIndex.value;
```
[the remark this answers is not in the document]
Comment (elliette) [reply packages/devtools_app/lib/src/ui/search.dart]: Done

File: packages/devtools_app/lib/src/ui/search.dart [hunk 4]
```diff
@@ -456,7 +501,16 @@ mixin AutoCompleteSearchControllerMixin on SearchControllerMixin {
...
+
+  set currentDefaultIndex(int index) {
+    _currentDefaultIndex.value = index;
+  }
```
[the remark this answers is not in the document]
Comment (elliette) [reply packages/devtools_app/lib/src/ui/search.dart]: Done

**Review by kenzieschmoll — APPROVED** [review]

**Pull request merged**